backend/src/LibExecution/ProgramTypes.fs

```diff
@@ -231,31 +288,27 @@ module DB =
       cols : List<Col> }
 
 module UserType =
-  type RecordField = { name : string; typ : Option<DType>; nameID : id; typeID : id }
-  type Definition = Record of List<RecordField>
+  // TODO: move this to some ComplexType or CustomType type
+  type RecordField = { id : id; name : string; typ : DType }
 
-  type T =
-    { tlid : tlid
-      name : string
-      nameID : id
-      version : int
-      definition : Definition }
+  type EnumField = { id : id; type_ : DType; label : Option<string> }
```

Comment:
What's the semantics here? Is it like:

```
type X1 =
  | Y1 of { a1 : int; b1 : int }
```

```
type X2 =
  | Y2 of a2 : int * b2 : int
```

or is there so difference there?

Reply:
I've made the labels on enum fields optional, so it's more like the second option here. If you'd rather them be required, I could update there, in which case the first option could be a fair representation later. Otherwise I think they're equivalent if labels are required.